interp/cfg.go

```diff
@@ -1511,6 +1511,18 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 			case isStruct(n.typ) || isInterfaceSrc(n.typ):
 				// Find a matching field.
 				if ti := n.typ.lookupField(n.child[1].ident); len(ti) > 0 {
+					if isStruct(n.typ) {
+						// If a method of the same name exists, use it if it is less shallowed.
+						// if method's depth is the same as field's, this is an error.
+						d := n.typ.methodDepth(n.child[1].ident)
+						if d >= 0 && d < len(ti) {
+							goto tryMethods
```

Comment:
why not 'goto default', since we want to fallthrough in default anyway? is it forbidden to use goto without a custom label?

Reply:
`default` is not usable as a label in Go, we need a custom label here.  I have not seen a better way to avoid a lot of code duplicates.

Reply:
ACK